examples/light_sense.rs

```diff
@@ -0,0 +1,223 @@
+        enable:                      false                      /* Not applicable, lesense controls this. */
+    };
+
+    let acmp = acmp::Acmp::acmp0();
+    /* Initialize ACMP */
+    acmp.init(&acmp_init);
+    /* Disable ACMP0 out to a pin. */
+    acmp.gpio_setup(0, false, false);
+    /* Set up ACMP negSel to VDD, posSel is controlled by LESENSE. */
+    acmp.channel_set(acmp::Channel::VDD, acmp::Channel::_0);
+    /* LESENSE controls ACMP thus ACMP_Enable(ACMP0) should NOT be called in order
+    * to ensure lower current consumption. */
+}
+
+fn setup_gpio() {
+    /* Configure the drive strength of the ports for the light sensor. */
+    gpio::drive_mode_set(LIGHTSENSE_EXCITE_PORT, gpio::DriveMode::Standard);
+    gpio::drive_mode_set(LIGHTSENSE_SENSOR_PORT, gpio::DriveMode::Standard);
+    /* Initialize the 2 GPIO pins of the light sensor setup. */
+    gpio::pin_mode_set(LIGHTSENSE_EXCITE_PORT, LIGHTSENSE_EXCITE_PIN, gpio::Mode::PushPull, 0);
+    gpio::pin_mode_set(LIGHTSENSE_SENSOR_PORT, LIGHTSENSE_SENSOR_PIN, gpio::Mode::Disabled, 0);
+    /* Configure user led as output */
+    gpio::pin_mode_set(LED_GPIO_PORT, LED_GPIO_PIN, gpio::Mode::PushPull, 0);
+}
+
+fn setup_rtc() {
+    let rtc_init = rtc::Init { enable: false, .. Default::default() };
+    rtc::init(&rtc_init);
+    /* Set compare value */
+    rtc::compare_set(0, RTC_COMP_VALUE);
+
+    rtc::int_enable((0x1 << 1)); // RTC_IFS_COMP0
+    nvic::enable_irq(nvic::IRQn::RTC);
+}
+
+fn setup_lesense() {
+    // LESENSE configuration structure
+    let init_lesense = lesense::Init {
+        // LESENSE configured for periodic scan.
+        core_ctrl: lesense::CoreCtrlDesc {
+            store_scan_res: true,
+            bias_mode:      lesense::BiasMode::DutyCycle, /* Lesense should duty cycle comparator and related references etc. */
+            debug_run:      false,
+            .. Default::default()
+        },
+
+        time_ctrl: Default::default(),
+
+        per_ctrl: lesense::PerCtrlDesc {
+            dac_ref:           lesense::DACRef::BandGap,
+            acmp0_mode:        lesense::ControlACMP::MuxThres, /* Allow LESENSE to control ACMP mux and reference threshold. */
+            acmp1_mode:        lesense::ControlACMP::MuxThres,
+            warmup_mode:       lesense::WarmupMode::Normal,    /* Normal mode means LESENSE is allowed to dutycycle comparator and reference. */
+            .. Default::default()
+        },
+
+        dec_ctrl: lesense::DecCtrlDesc {
+            hyst_prs0:   false,
+            hyst_prs1:   false,
+            hyst_prs2:   false,
+            hyst_irq:    false,
+            prs_count:   true,
+            .. Default::default()
+        }
+    };
+
+    /* Channel configuration */
+    /* Only one channel is configured for the lightsense application. */
+    let init_lesense_ch = lesense::ChDesc {
```

Comment:
It could, but I chose to set every field explicitly in order to keep the comments that describes the values of each field.